File: Fugazi/src/org/fugazi/ql/ast/expression/literal/Literal.java
package org.fugazi.ql.ast.expression.literal;

import org.fugazi.ql.ast.expression.Expression;
import org.fugazi.ql.ast.expression.IExpressionVisitor;

public abstract class Literal extends Expression {

    public Literal() {
        super();
    }

    public abstract String toString();

    public abstract <T> T accept(IExpressionVisitor<T> visitor);
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Literals/Literal.cs
using AST.Nodes.Expressions;
using AST.Nodes.Interfaces;

namespace AST.Nodes.Literals
{
    public abstract class Literal : Expression, IHasType
    {
        protected Literal(PositionInText position)
            : base(position) {}
        public abstract Types.Type RetrieveType();

        public abstract override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor);
    } 
}
